docker/Dockerfile_full

```diff
@@ -24,41 +29,58 @@ RUN apt-get update -qq && \
   libffi-dev \
   libpng-dev \
   libpq-dev \
-  curl && \
-  apt-get clean && \
-  rm -rf /var/lib/apt/lists/* /tmp/* /var/tmp/* && \
-  mkdir /install && \
-  mkdir /app
+  curl
 
-WORKDIR /install
+# Install spacy models
+RUN pip install https://github.com/explosion/spacy-models/releases/download/en_core_web_md-2.1.0/en_core_web_md-2.1.0.tar.gz#egg=en_core_web_md==2.1.0 --no-cache-dir > /dev/null \
+    && python -m spacy link en_core_web_md en \
+    && pip install https://github.com/explosion/spacy-models/releases/download/de_core_news_sm-2.1.0/de_core_news_sm-2.1.0.tar.gz#egg=de_core_news_sm==2.1.0 --no-cache-dir > /dev/null \
+    && python -m spacy link de_core_news_sm de
 
-# Copy as early as possible so we can cache ...
+# Copy only what we really need
+COPY README.md .
+COPY setup.py .
+COPY MANIFEST.in .
 COPY alt_requirements/ ./alt_requirements
 COPY requirements.txt .
 
+# Install Rasa and its dependencies
 RUN pip install -r alt_requirements/requirements_full.txt
 
-COPY --from=builder /build/rasa.tar.gz .
-RUN pip install ./rasa.tar.gz[sql,spacy,mitie]
+COPY rasa ./rasa
+
+# Install Rasa as package
+COPY rasa ./rasa
+RUN pip install .
 
+# Runtime stage which uses the virtualenv which we built in the previous stage
+FROM base AS runner
+
+# Install mitie model
 RUN apt-get update -qq \
     && apt-get install -y --no-install-recommends wget \
     && wget -P /app/data/ https://s3-eu-west-1.amazonaws.com/mitie/total_word_feature_extractor.dat \
     && apt-get remove -y wget \
     && apt-get autoremove -y
 
-RUN pip install https://github.com/explosion/spacy-models/releases/download/en_core_web_md-2.1.0/en_core_web_md-2.1.0.tar.gz#egg=en_core_web_md==2.1.0 --no-cache-dir > /dev/null \
-    && python -m spacy link en_core_web_md en \
-    && pip install https://github.com/explosion/spacy-models/releases/download/de_core_news_sm-2.1.0/de_core_news_sm-2.1.0.tar.gz#egg=de_core_news_sm==2.1.0 --no-cache-dir > /dev/null \
-    && python -m spacy link de_core_news_sm de
-
+# Copy over default pipeline config
 COPY sample_configs/config_pretrained_embeddings_spacy_duckling.yml /app/config.yml
 
-VOLUME ["/app"]
+# Copy virtualenv from previous stage
+COPY --from=builder /build /build
+
 WORKDIR /app
 
+# Create a volume for temporary data
+VOLUME /tmp
+
+# Make sure the default group has the same permissions as the owner
```

Comment:
could create a group and a user then changing permissions:
USER rasauser
RUN mkdir -p $APPDIR && \
        addgroup -g 10001 rasauser && \
        adduser -S -G rasauser -u 10001 -h $APPDIR rasauser && \
        chown -R rasauser $APPDIR

Reply:
For openshift we required it that everything is accessible by the group. The user is random anyway. So I think the current way is ok 🤔